src/engraving/libmscore/tie.cpp

```diff
@@ -117,7 +117,7 @@ void TieSegment::draw(mu::draw::Painter* painter) const
 bool TieSegment::isEditAllowed(EditData& ed) const
 {
     if (ed.key == Key_X && !ed.modifiers) {
-        return true;
+        return false;
```

Comment:
actually I have no idea why this code is here at all, 'x' already maps to the 'flip' action which works as expected when a tie is selected.

Reply:
If it works when returning false, yes, then let's remove the whole special case. In that case, please also remove the corresponding code in `TieSegment::edit`, `SlurSegment::isEditAllowed` and `SlurSegment::edit`. 

So now I finally understand exactly what's happening: both "Slur" and "Tie" have this strange special case for flipping using the X key, probably it was necessary at some point in history, but "Slur" correctly uses `undoChangeProperty` but "Tie" uses `setSlurDirection` which skips the undo stack and also fails to update the layout. So that's why it does work for Slur but not for Tie!